### DISCUSSION TITLE
Bundling Classical Scripts

### DISCUSSION THREAD

**Discussion by CxRes** [parcel-bundler/parcel#8398] [Q&A]
Is there a way to bundle a bunch of classic scripts using Parcel?

I am dealing with a bunch of scripts that are being distributed as IIFE's with no way to access module source. I would like to combine them into a single script that I can host myself:

Something like:
```html
<script src="npm:docsify"></script>
<script src="npm:docsify-themeable"></script>
```

into:
```html
<script src="/bundle.hash.js"></script>
```

As you can see above in my specific case I wish to bundle Docsify and its plugins into a single file that I can self host....

**Comment by mischnic** [comment]
> combine them into a single script 

Combining multiple script tags into a single output bundle isn't easily possible right now.

A `<script src="bundle.js></script>` where bundle.js contains multiple JS imports `import "docsify";` would certainly work.

**Reply by CxRes** [reply]
These scripts are pre-built; they cannot be "required" or "imported". That's why I had specifically asked about classical scripts (in implementation terms it is a simple matter of dumping their contents in one file, but Parcel appears to have no mechanism for this). Should I put in a feature request?

**Reply by mischnic** [reply]
Ah, so you really need them to be `<script>` and not `<script type="module">`

> in implementation terms it is a simple matter of dumping their contents in one file

Yes, it's simple. But this is a very rare usecase

**Reply by CxRes** [reply]
> Yes, it's simple. But this is a very rare usecase.

I am actually a bit surprised by that. If one is self hosting a website, combining scripts into a single file (even before a module system) seems to be a rather obvious optimization to have.
